### DISCUSSION TITLE
Manually adding Artist, Album, Title

### DISCUSSION THREAD

**Discussion by JoeyMudflats** [enzo1982/freac#323] [General Discussion]
Looking for a CD Ripper to catalogue CD audiobooks, which don't have existing entries in CDDB.
Fre:ac seems to be much more powerful - and twice as fast - as Sound Juicer for instance, in Linux.
However, audiobook CDs need manual entry of Album, Artist, Title info but I can't find a way to enter this information for all the tracks on a CD except to enter the info laboriously for each track. Have I missed something? Is there a quick way to replicate this info from track 1 to all the other tracks?
Fre:ac is clearly superb software, thanks to Robert for his hard work.

**Comment by Lakrits** [comment]
In order to copy e.g. the artist for a track to all other tracks (or, rather, all other selected tracks), you can do the following:

1. Select the tracks you want to copy the information to. The checkboxes are found to the left of the posts in the job-list.

![1. Checkboxes](https://user-images.githubusercontent.com/1184718/173367932-271b68b9-2118-4114-ae66-533300fb1e95.png)

2. In the job-list, select the track holding the information you want to replicate. Then move the cursor to the top right corner of the field containing that information. You'll see a small triangle popping up, indicating a drop-down selection menu. These menus are available for all information fields, except for the field "Track".
 
![2. Menu indicator](https://user-images.githubusercontent.com/1184718/173368510-9217acc3-fbdc-4c06-b2b5-f1d211bc51db.png)

3. In the drop-down menu, select the option "Use for all selected tracks".

![3. Use for all selected tracks](https://user-images.githubusercontent.com/1184718/173369936-20723a9c-e079-40ea-ac94-353a29048347.png)

4. That's all. All the selected tracks now share that piece of information.

![4. Information copied](https://user-images.githubusercontent.com/1184718/173370551-35330b1a-00cf-4bee-9cb5-fa79c3ebab9d.png)

**Comment by seamusdemora** [comment]
@Lakrits : This is cool & useful, BUT: 

1. It doesn't work for cover art > no similar technique to automatically add cover art to all tracks 

2. It doesn't extend to many of the tags; e.g. Orchestra, Conductor, Composer, etc : 

what you create for track 1:
![Screenshot-4](https://user-images.githubusercontent.com/14911811/186995349-f448ceab-2096-42be-8610-4f5c1008a61d.png) 

cannot be replicated for tracks 2-6:
![Screenshot-6](https://user-images.githubusercontent.com/14911811/186995547-fdcbc19a-a8f9-48b5-9381-9669573f8fea.png)

Leaving many "blanks" to be filled in manually: 
![Screenshot-1](https://user-images.githubusercontent.com/14911811/186995907-695ae7dd-f74f-439c-9ed8-fa67183a64bf.png)

3. Am I missing something??

**Reply by strange-antiquark** [reply]
Have you tried Tags/Album? I'm pretty sure you should be able to do the composer for the entire album there.

**Reply by enzo1982** [reply]
Yes, @strange-antiquark is right. You should be able to edit the orchestra, conductor, composer and also cover images for the whole album at once on the _Tags->Album_ tab.

**Reply by seamusdemora** [reply]
Thanks! I may have missed that. It wasn't clear to me that it would apply to all tracks on the album.  In fact, if you look at my final screenshot above - that more or less convinced me that this was not working. However, this was my first use with `fre:ac`, and I cannot recall the exact sequence that I went through. I did successfully convert from my CD to FLAC files (they sound wonderful), and I then began surveying what I had in terms of the metadata.

And so, I am now sitting on the `Tags->Album - Details` screen as shown below. I have again entered the Composer name. But when I look back at the `Joblist` screen, the composer's name appears only on the first track. What must I do next to update my FLAC files? Must I start from the beginning - from the CD??

![Screenshot from 2022-08-28 18-41-43](https://user-images.githubusercontent.com/14911811/187089948-035b68ff-6e7d-4b78-b146-25e0da452006.png)

I do hope this next question doesn't put anyone off, but:   Do you think that editing the metadata may be more clearly presented as a separate task from the encoding? Using this CD as an example: It is 3 symphonies on 2 CDs. It doesn't strike me that this is the proper way to organize the files in my library. Instead of a folder for each CD, I'd like 3 folders - one for each symphony - and track #'s according to the symphony instead of the CD. Again, I don't want to sound presumptuous - I'm interested in some feedback from experienced users & developers.

**Reply by strange-antiquark** [reply]
> And so, I am now sitting on the Tags->Album - Details screen as shown below. I have again entered the Composer name. But when I look back at the Joblist screen, the composer's name appears only on the first track.

That's because you made the edit in Tags/Tracks. For some reason, that information also appears in Tags/Albums, but it hasn't actually changed anything for the other tracks.
Go back to Tags/Tracks and remove the composer name for the track, then go back to Tags/Albums and enter it there. The composer name will now show up on all tracks.

> I do hope this next question doesn't put anyone off, but: Do you think that editing the metadata may be more clearly presented as a separate task from the encoding? Using this CD as an example: It is 3 symphonies on 2 CDs... I'd like 3 folders - one for each symphony....

This should probably be a separate thread, but.... this is just one way to do it, but it is the simplest way I believe.
Since you need to keep the first CD in the drive until the tracks on it are written to your HDD, this will have to be done in 2 stages. Read in CD1, then de-select all the tracks belonging to the second symphony. Now edit the Album field for one movement of the first, then use the identical process for the Album field that Lakrits mentioned for the Artist field. The button to do so is in the same location, but on the Album row. Now "toggle" the entries, and do the same for the second symphony. You may edit the track numbers for the second symphony at this time also.
Now go do Tags/Albums, and you will find you have two albums, one for each symphony. Edit the tracks fields here, the number of movements in each symphony. You may also edit the Disc fields (both number and total) if you wish, or you may remove them entirely -- your choice.
Now you should be ready to write all these tracks to your drive, so back to the Joblist and "select all" -- probably a good idea to see that everything is what you want, and it's going to the right place before you begin processing, though.
After the first CD is processed, load up the second one, and perform the same procedure on it as you just did for the first. Be sure to use the same Album title for 2nd symphony movements as you used for CD1, to ensure all its movements are written to the same folder.

Oh, I almost forgot... where does the output go? I would suggest using an output folder something like "Tchaikovsky - Symphonies 4, 5 and 6" (or whatever else you may wish), and set the "output filenames" to (as a minimum) "\<album\>/\<track\>. \<title\>"
Also note that, for works completed in different years, you can also edit the years field in the Joblist, and again, apply it to all selected tracks, as per Lakrit's method above.

**Comment by seamusdemora** [comment]
Thanks for that explanation. It all sounds a bit complex, and as I'll never be a frequent user, perhaps I'll try to find a dedicated metadata editor.

**Reply by strange-antiquark** [reply]
I hear you.
Recovering from a mis-step can be a tad difficult, but it all becomes must simpler, and less error-prone, if you follow one basic rule: Do all metadata edits that you can first on the entire album, then proceed to edit track-by-track.
